src/writer.rs

```diff
@@ -0,0 +1,58 @@
+use std::io::{Write, Error, ErrorKind};
+use std::io;
+use InflateStream;
+
+pub struct InflateWriter<W: Write> {
+    inflater: InflateStream,
+    writer: W
+}
+
+impl<W: Write> InflateWriter<W> {
+    pub fn new(w: W) -> InflateWriter<W> {
+        InflateWriter { inflater: InflateStream::new(), writer: w }
+    }
+
+    pub fn finish(mut self) -> io::Result<W> {
```

Comment:
I tried to use the same name as [flate2](https://docs.rs/flate2/0.2.17/flate2/write/struct.DeflateDecoder.html) and [deflate-rs](https://github.com/oyvindln/deflate-rs/blob/dev/src/writer.rs) (thanks @oyvindln for that great library!). Although I notice know that flate2 uses the signature `fn finish(self) -> Result<W>` and deflate-rs (and this patch) uses the signature `fn finish(mut self) -> Result<W>`. Any preferences? I don't have any strong opinion on this :smile: